ts/features/bonus/bonusVacanze/store/reducers/availableBonusesTypes.ts

```diff
@@ -65,14 +66,16 @@ export const visibleAvailableBonusSelector = createSelector(
     pot.getOrElse(
       pot.map(availableBonusesState, bonuses =>
         bonuses.filter(b => {
-          const isExperimentalEnabled =
-            fromNullable(mapBonusIdFeatureFlag.get(b.id_type)).getOrElse(
-              false
-            ) && b.visibility === BonusVisibilityEnum.experimental;
-
+          const isFeatureFlagEnabled = fromNullable(
+            mapBonusIdFeatureFlag().get(b.id_type)
+          ).getOrElse(false);
+          const isFFEnabledAndExperimental =
+            isFeatureFlagEnabled &&
+            b.visibility === BonusVisibilityEnum.experimental;
           return (
-            isExperimentalEnabled ||
-            b.visibility === BonusVisibilityEnum.visible
+            isFFEnabledAndExperimental ||
+            (isFeatureFlagEnabled &&
+              b.visibility === BonusVisibilityEnum.visible)
```

Comment:
What does makes this differ by `EXPERIMENTAL`?

Using this implementation wouldn't make `EXPERIMENTAL` value useless?

Reply:
this logic should cover these two cases:

A bonus is visibile when (at least one of these):

1. - the relative FF is enabled and the visibility is `experimental` 
1. - the relative FF is enabled and the visibility is `visible`

case 1
```typescript
const isFeatureFlagEnabled = fromNullable(
            mapBonusIdFeatureFlag().get(b.id_type)
          ).getOrElse(false);
const isFFEnabledAndExperimental =
isFeatureFlagEnabled &&
b.visibility === BonusVisibilityEnum.experimental;
```

case 2
```typescript
isFeatureFlagEnabled &&
              b.visibility === BonusVisibilityEnum.visible
```

case 1: it is useful when we release an internal beta with FF enabled: production can't view this bonus since it is experimental and FF off
case 2: we don't show a bonus if FF is off or visibility is `hidden` or `experimental`. 

The previous logic will include a bonus even if the FF is disabled 
```typescript
return (
            isExperimentalEnabled ||
            b.visibility === BonusVisibilityEnum.visible)
```

Reply:
Ok, so you confirmed my initial doubt 😄 

since visible and experimental are working in the same way we don't need both. Using your described cases:

- We deliver the production app with FF disabled, but it is remotely tagged as `visible` -> production doesn't show the bonus item
- We deliver the production app with FF enabled and `visible` value -> production app can see the item
- We deliver the beta app with FF enabled and `visible` value -> beta app can see the item
- Any cases if the value is `hidden` we hide the item

Am I some way confusing?